Publish A-102375 (#16925)
cockroachdb/docs: advisories/a102375.md

This fix will be applied to the maintenance release of CockroachDB v22.1.20.

This public issue is tracked by [#102375](https://github.com/cockroachdb/cockroach/issues/102375).

## Mitigation

Users of CockroachDB v22.1.19 and v22.2.8 who experience spurious [privilege](../v22.2/security-reference/authorization.html#privileges) errors with the query cache enabled are encouraged to upgrade to v22.1.20, v22.2.9, or a later version.

If an upgrade is not possible, the issue can be avoided by updating the SQL queries to qualify table names with the database name so there is no collision in the query cache. For example, `SELECT * FROM table_name` can be rewritten using [partially qualified](../v22.2/sql-name-resolution.html#lookup-with-partially-qualified-names) or [fully qualified](../v22.2/sql-name-resolution.html#lookup-with-fully-qualified-names) names as follows:

- `SELECT * FROM database_name.table_name`
- `SELECT * FROM database_name.schema_name.table_name`

Another option, if an upgrade is not possible, is to disable the query cache with the following command:

{% include_cached copy-clipboard.html %}
~~~ sql
SET CLUSTER SETTING sql.query_cache.enabled = false;
~~~

> However, note that this may degrade the performance of the cluster.

Reviewer: ```suggestion
Disabling the query cache may degrade the performance of the cluster, however.
```

## Impact

> Some customers running identical queries with different roles to access tables with the same name in different databases could experience spurious [privilege](../v22.2/security-reference/authorization.html#privileges) errors on CockroachDB v22.1.19 and v22.2.b8 with the query cache enabled.

Reviewer 2: v22.2.b8 -> v22.2.8

Author: fixed, thanks!
